crates/client/rpc/src/methods/trace/simulate_transactions.rs

```diff
@@ -0,0 +1,193 @@
+        BroadcastedTransaction::DeployAccount(deploy_account_tx) => {
+            deploy_account_tx.try_into().map(|tx| (TxType::DeployAccount, tx))
+        }
+    });
+    let (tx_types, user_transactions) =
+        itertools::process_results(tx_type_and_tx_iterator, |iter| iter.unzip::<_, _, Vec<_>, Vec<_>>()).map_err(
+            |e| {
+                error!("Failed to convert BroadcastedTransaction to UserTransaction: {e}");
+                StarknetRpcApiError::InternalServerError
+            },
+        )?;
+
+    let simulation_flags = SimulationFlags::from(simulation_flags);
+
+    let res = starknet
+        .client
+        .runtime_api()
+        .simulate_transactions(substrate_block_hash, user_transactions, simulation_flags)
+        .map_err(|e| {
+            error!("Request parameters error: {e}");
+            StarknetRpcApiError::InternalServerError
+        })?
+        .map_err(|e| {
+            error!("Failed to call function: {:#?}", e);
+            StarknetRpcApiError::ContractError
+        })?;
+
+    let storage_override = starknet.overrides.for_block_hash(starknet.client.as_ref(), substrate_block_hash);
+    let simulated_transactions =
+        tx_execution_infos_to_simulated_transactions(&**storage_override, substrate_block_hash, tx_types, res)
+            .map_err(StarknetRpcApiError::from)?;
+
+    Ok(simulated_transactions)
+}
+
+fn tx_execution_infos_to_simulated_transactions<B: BlockT>(
+    storage_override: &dyn StorageOverride<B>,
+    substrate_block_hash: B::Hash,
+    tx_types: Vec<TxType>,
+    transaction_execution_results: Vec<
+        Result<TransactionExecutionInfo, PlaceHolderErrorTypeForFailedStarknetExecution>,
+    >,
+) -> Result<Vec<SimulatedTransaction>, ConvertCallInfoToExecuteInvocationError> {
+    let mut results = vec![];
+    for (tx_type, res) in tx_types.into_iter().zip(transaction_execution_results.into_iter()) {
+        match res {
+            Ok(tx_exec_info) => {
+                let transaction_trace =
+                    tx_execution_infos_to_tx_trace(storage_override, substrate_block_hash, tx_type, &tx_exec_info)?;
+                let gas_consumed =
+                    tx_exec_info.execute_call_info.as_ref().map(|x| x.execution.gas_consumed).unwrap_or_default();
+                let overall_fee = tx_exec_info.actual_fee.0 as u64;
+                // TODO: Shouldn't the gas price be taken from the block header instead?
+                let gas_price = if gas_consumed > 0 { overall_fee / gas_consumed } else { 0 };
+
+                results.push(SimulatedTransaction {
+                    transaction_trace,
+                    fee_estimation: FeeEstimate { gas_consumed, gas_price, overall_fee },
+                });
+            }
+            Err(_) => {
+                return Err(ConvertCallInfoToExecuteInvocationError::TransactionExecutionFailed);
+            }
+        }
+    }
+
+    Ok(results)
+}
+
+// fn try_get_function_invocation_from_call_info(
```

Comment:
Why is this commented code here? Maybe consider removing it if it non longer serves a purpose.